uber/models.py

```diff
@@ -1297,6 +1227,23 @@ def _staffing_adjustments(self):
         self.trusted_depts = ','.join(str(td) for td in self.trusted_depts_ints if td in self.assigned_depts_ints)
 
     @presave_adjustment
+    def _badge_adjustments(self):
+        # _assert_badge_lock()
+        from uber.badge_funcs import needs_badge_num
+        if self.badge_type == c.PSEUDO_DEALER_BADGE:
+            self.ribbon = c.DEALER_RIBBON
+
+        self.badge_type = get_real_badge_type(self.badge_type)
+
+        if not needs_badge_num(self):
+            self.badge_num = None
+
+        if self.orig_value_of('badge_type') != self.badge_type or self.orig_value_of('badge_num') != self.badge_num:
+            self.session.update_badge(self, self.orig_value_of('badge_type'), self.orig_value_of('badge_num'))
```

Comment:
This is what calls update_badge() for our change_badge page now. Plus, thanks to our setting the badge_num to `None` just above, this will also call update_badge for badges that are given INVALID status or changed by the attendee (which can be done in rare cases) or by other functions (like `unset_volunteering`, which is now called BEFORE this and does not itself call update_badge()).